starboard/nplb/posix_compliance/posix_pipe_test.cc

```
@@ -31,14 +31,27 @@ namespace {
 
 // Only used by the WriteToFullPipeWithNonBlockFlagImmediatelyReturnsError test.
 struct TestContext {
-  TestContext(int read_fd, int timeout_s)
-      : out_of_time(false), read_fd(read_fd), timeout_s(timeout_s) {
+  TestContext(int read_fd, int64_t timeout_duration_us)
+      : out_of_time(false),
+        had_expected_failed_write(false),
+        read_fd(read_fd),
+        timeout_duration_us(timeout_duration_us) {
     EXPECT_EQ(pthread_mutex_init(&out_of_time_mutex, NULL), 0);
+    EXPECT_EQ(pthread_mutex_init(&had_expected_failed_write_mutex, NULL), 0);
+    EXPECT_EQ(pthread_cond_init(&had_expected_failed_write_cv, NULL), 0);
   }
+
+  // Whether the timeout has been reached.
   bool out_of_time;
   pthread_mutex_t out_of_time_mutex;
+
+  // Whether the main thread returned from a failed write to a full pipe.
+  bool had_expected_failed_write;
```

I find the expected_failed name to be confusing—what do you think about a name like `bool blocked_write`?

---

Can you please explain what seems confusing about it? The name is longer than I would like, especially when we add the `_mutex` and `_cv` suffixes, but I think it really captures what we want: we expect the `write()` call to eventually fail in a particular (`EAGAIN`) way, rather than fail in an unexpected way (with another error) or to fail to fail because the function is blocked from returning.

Would you prefer `avoided_blocked_write`? I'd also be happy with this name.

I'd rather not use `blocked_write` for a couple reasons:
1. AFAIK we don't have a way to definitively say that the write is blocked. The best we can do is estimate that it's blocked, from the `DoReadOnTimeout` thread, based on other information.
2. I think the use of this variable would be more confusing, at least without redesigning the test. I'm assuming we would initialize it to true and then set it to false in the main thread when the `write()` call fails with `EAGAIN`? That seems confusing since we'd be setting `blocked_write` to true before we even suspect that writing is blocked.